content/articles/integrating-facebook-ads-into-your-android-application-in-android-studio/index.md

```diff
@@ -0,0 +1,186 @@
+android:layout_width="match_parent"
+android:layout_height="match_parent"
+tools:context=".MainActivity">
+ 
+<LinearLayout
+android:id="@+id/banner_container"
+android:layout_width="match_parent"
+android:layout_height="wrap_content"
+android:layout_alignParentBottom="true"
+android:orientation="vertical"
+App:layout_constraintBottom_toBottomOf="parent"/>
+ 
+</RelativeLayout>
+
+```
+
+Implement the Banner Ad by instantiating an Adview object on your java activity. To do that, place this code in the `onCreate()` function.
+
+```kotlin
+adView = new AdView(this, "IMG_16_9_App_INSTALL#2029572424039676_2029575434039375", AdSize.BANNER_HEIGHT_60);
+ 
+// Find the AdContainer
+LinearLayout adContainer = (LinearLayout) findViewById(R.id.banner_container);
+ 
+adContainer.addView(adView);
+ 
+// Load an ad
+adView.loadAd();
+
+```
+Paste the following code in your activity's `onDestroy()` function. This is done to free up AdView's resources.
+
+```kotlin
+@Override
+protected void onDestroy() {
+if (adView != null) {
+adView.destroy(); //checks an ad, if the ad is not present(null) the view is destroyed
+}
+super.onDestroy();
+}
+
+```
+
+With all that done, run your code and observe the output.
+For successful integration, the following output is obtained.
+
+![output](engineering-education/integrating-facebook-ads-with-your-android-application-in-android-studio
+/output.png)
+
+You can alter the code to and other types of Ads like interstitial Ads and Native Ads in the same manner.
+
+### Why integrate Facebook ads?
+Consumers prefer in-App advertising over display advertising on mobile websites. Smartphone consumers find ads on the mobile web to be compacted and unpleasant. In-App advertisements are resized to fit the screen and seem better, increasing overall engagement.
+
+When establishing in-App advertisements, **accessibility** is another important issue to consider. In-App mobile advertising is easier to reach because most consumers have their phones with them all the time, wherever they go. As a result, you have access to a captive audience or users already using the App and are interested in what you have to say. It's easier to convert them because they're already interested.
+
+### Demerits of integrating in-app ads into some Applications
+In-App advertisement, as seen earlier, has many advantages. Besides, it is not always Appealing to some users when ads keep popping in all the time.
+
+Here are some demerits in an in-App advertisement;
+
+- **Disturbances from ads pop-up** create a user's negative view towards the application. This may discourage users from using applications known to have in-App advertisements.
+
+- **Difficult to navigate** - Some interested customers may not get to see the advertisement due to the incompatibility of the ads with some phones.
+
+- **Cost incurred on the user** to get the advertisement. To get the advertisement, the user's internet bundles get consumed from an unintended advertisement viewing task.

```

Comment:
This point isn't clear. The user's internet bundle? If you are referring to cellular data, it may be better to reword it.